src/renderers/__tests__/refs-test.js

```diff
@@ -476,3 +476,42 @@ describe('root level refs', () => {
     }
   });
 });
+
+describe('creating element with ref in constructor', () => {
+  class RefTest extends React.Component {
+    constructor(props) {
+      super(props);
+      this.p = <p ref="p">Hello!</p>;
+    }
+
+    render() {
+      return <div>{this.p}</div>;
+    }
+  }
+
+  it('throws an error when __DEV__ = true', () => {
+    ReactTestUtils = require('react-dom/test-utils');
+
+    var originalDev = __DEV__;
+    __DEV__ = true;
+
+    expect(function() {
+      ReactTestUtils.renderIntoDocument(<RefTest />);
+    }).toThrowError();
+
+    __DEV__ = originalDev;
+  });
+
+  it('throws an error when __DEV__ = false', () => {
+    ReactTestUtils = require('react-dom/test-utils');
+
+    var originalDev = __DEV__;
+    __DEV__ = false;
+
+    expect(function() {
+      ReactTestUtils.renderIntoDocument(<RefTest />);
+    }).toThrowError();
```

Comment:
I originally tried checking for the specific error message. I copy/pasted the message but it said actual vs. expected message didn't match, even though they appeared identical. I'll give it another try.